libs/cardano-ledger-binary/test/Test/Cardano/Ledger/Binary/RoundTripSpec.hs

```diff
@@ -77,6 +97,30 @@ import Test.Cardano.Ledger.Binary.Arbitrary ()
 import Test.Cardano.Ledger.Binary.RoundTrip
 import Test.Hspec
 
+withSK :: KESAlgorithm v
+       => PinnedSizedBytes (SeedSizeKES v) -> (SignKeyKES v -> IO b) -> IO b
+withSK seedPSB action =
+  bracket
+    (fmap MLockedSeed . mlsbFromByteString . psbToByteString $ seedPSB)
+    mlockedSeedFinalize
+    $ \seed ->
+      bracket
+        (genKeyKES seed)
+        forgetSignKeyKES
+        action
+
+mkVerKeyKES :: KESAlgorithm v
+            => PinnedSizedBytes (SeedSizeKES v)
+            -> IO (VerKeyKES v)
+mkVerKeyKES seedPSB =
+  withSK seedPSB deriveVerKeyKES
+
+mkSigKES :: (KESAlgorithm v, ContextKES v ~ (), Signable v BS.ByteString)
+         => (PinnedSizedBytes (SeedSizeKES v), [Word8])
+         -> IO (SigKES v)
+mkSigKES (seedPSB, msg) =
+  withSK seedPSB $ \sk -> (signKES () 0 (BS.pack msg) sk)
+
```

Comment:
All KES tests can be removed. Again. We don't need them, if we do not provide those serialization instances

Reply:
I believe we do need ser/deser on verification keys, because that is used to ser/deser OCerts. And in order to generate a verification key for testing, we need to derive it from a signing key, so we need at least `withSK`.

Reply:
I think you missed my point. We need to remove all this new KES keys serialization related functionality from `cardano-ledger-binary`, because ledger does not rely on it.  The fact that OCert contains KES does not interfere with that point, because the `cardano-protocol-tpraos` is really a consensus package. We hope to move it some day into the consensus repo. We can continue using the serialization mechanism provided by `cardano-crypto-class` and `cardano-binary` in those places where KES serialization is needed.

In places like `OCert` we can rely on serialization provided by unsound serialization, it just doesn't need to come from `cardano-ledger-binary`. All you need is `fromPlainDecoder` to lift the regular `Decoder` from `cborg` to the `Decoder` provided by the `cardano-ledger-library` package.

https://github.com/IntersectMBO/cardano-ledger/blob/d24211a6d8a04222a7b9add7a1d9157f2d631933/libs/cardano-protocol-tpraos/src/Cardano/Protocol/TPraos/OCert.hs#L135-L137

The only reason why serialization for KES was provided by `cardano-ledegr-binary` was for completeness. In other words we do not need to provide unsound support, since it will be removed at some point later.

Reply:
So, we can continue using sound support for SigKES and for VerKeyKES. While *unsound* support for SignKeyKES remains to be restricted to cardano-base, with some usage of it in ledger, which is restricted to testing.